addons/website_sale/tests/test_sale_process.py

```diff
@@ -196,12 +199,7 @@ def test_02_demo_address_and_company(self):
             self.assertTrue(new_shipping.company_id != self.env.user.company_id, "Logged in user new shipping should not get the company of the sudo() neither the one from it's partner..")
             self.assertEqual(new_shipping.company_id, self.website.company_id, ".. but the one from the website.")
 
-            # 2. Logged in user, edit billing
-            self.default_address_values['partner_id'] = self.demo_partner.id
-            # Name cannot be changed if there are issued invoices
-            self.default_address_values['name'] = self.demo_partner.name
-            self.WebsiteSaleController.address(**self.default_address_values)
-            self.assertEqual(self.demo_partner.company_id, self.company_c, "Logged in user edited billing (the partner itself) should not get its company modified.")
+            # 2. Logged in user/internal user, should not edit billing
```

Comment:
As discussed previously.... you miss here the more important test ! 
Check that the demo partner is not able to change his email...

```python
  # 2. Logged in user (employee), edit billing
  self.default_address_values['partner_id'] = self.demo_partner.id
  self.default_address_values['name'] = 'new name'
  self.default_address_values['email'] = 'new_email@example.com'
  self.default_address_values['partner_id'] = self.demo_partner.id
  self.WebsiteSaleController.address(**self.default_address_values)
  self.assertNotEqual(
      self.demo_partner.email,
      'new_email@example.com',
      "Employee cannot change their email during the checkout process."
  )
  self.assertNotEqual(
      self.demo_partner.name,
      'new name',
      "Employee cannot change their name during the checkout process."
  )
```